### PR TITLE
docs: add line on installing xcode for macOS build flow

### PR DESCRIPTION
**Pull request by LisaLudique**
Signed-off-by: Lisa Lu <lisalu@lyft.com>

Description: Because of rules_foreign_cc in bazelbuild, Envoy will not compile successfully when following the instructions in the build docs due to how the tools are referenced. One fix for this is installing Xcode from the App Store (see https://github.com/bazelbuild/rules_foreign_cc/issues/185).

Risk Level: Low
Testing: N/A (docs change)
Docs Changes: see Description
Release Notes: N/A

### CODE REVIEW

File: ci/README.md [hunk 1]
```diff
@@ -133,7 +133,8 @@ The macOS CI build is part of the [CircleCI](https://circleci.com/gh/envoyproxy/
 Dependencies are installed by the `ci/mac_ci_setup.sh` script, via [Homebrew](https://brew.sh),
 which is pre-installed on the CircleCI macOS image. The dependencies are cached are re-installed
 on every build. The `ci/mac_ci_steps.sh` script executes the specific commands that
-build and test Envoy.
+build and test Envoy. If Envoy cannot be built (`error: /Library/Developer/CommandLineTools/usr/bin/libtool: no output file specified (specify with -o output)
+`), ensure that XCode is installed.
```
Comment (zuercher) [inline ci/README.md]: nit: remove the newline here and wrap after the comma

**Comment by zuercher** [comment]
You'll have to edit/force-push that last commit with the DCO signoff. (I think `git commit --amend --no-edit --signoff && git push -f` will do the trick.)

**Comment by zuercher** [comment]
Still missing DCO on 46beec.

**Comment by LisaLudique** [comment]
I think that did the trick, thanks!

**Pull request merged**